draft-ion.adoc

```diff
@@ -208,79 +200,176 @@ represented as follows:
+
+If a member name is not equal to the case-sensitive octet sequence `meta`, an Ion parser MUST NOT identify a member as a _Meta Object_.
+
+The member names within an identified _Meta Object_ MUST be unique; Ion parsers MUST either
+reject _Meta Object_ members with duplicate member names or use a JSON
+parser that returns only the lexically last duplicate member, as specified
 in {uri-ecmaScript-5-1-15-12}[Section 15.12 (The JSON Object)] of {uri-ecmaScript-5-1}[ECMAScript 5.1].
 
-The set of fields that a _Meta Object_ must contain to be considered valid
+The set of members that a _Meta Object_ must contain to be considered valid
 is resource-dependent.  In the absense of any resource-specific requirements,
-any field not understood by implementations MUST be ignored.
+any member not understood by implementations MUST be ignored.
+
+[[links]]
+== Links
+
+An Ion Link is an Ion Object where its `meta` member contains an {uri-IRIs}[IRI] member named `href`. An Ion Link is therefore JSON serialization of a {uri-rfc5988}[web link] to another resource.
+
+Ion parsers MUST identify any JSON structure matching this definition as an Ion Link.  Anything that does not match this defined structure MUST NOT be considered an Ion Link.
+
+A non-normative example of a minimal Ion link:
+
+.Example simplest Ion link:
+[source,json]
+----
+{ "meta": { "href": "https://api.example.com/corporations/acme" } }
+----
+
+=== Link Relation Types
+
+All Ion links have one or more <<links, link relation types>>.  A link's total set of link relation types is the set union of the link's implicit link relation type and the link's explicit link relation types.
+
+==== Implicit Link Relation Type
+
+All Ion links have exactly one implicit link relation type.  The implicit link relation type can be one of three possible values depending on where the link is located in the Ion content:
+
+1. Root Object
+2. Named Link
+3. Collection Resource `items` array element
+
+===== Root Object Implicit Link Relation Type
+
+If the Root Object is also found to be an Ion Link, it has an implicit link relation type of `self`.
+
+A non-normative example of a Root Object that is also an Ion Link:
+
+[source,json]
+----
+{
+  "meta": { "href": "https://api.example.com/users/1" },
+  "name": "Joe"
+}
+----
 
-=== Registered Metadata Field Names
+===== Named Link Implicit Link Relation Type
 
-The following field names are registered in the IANA Ion Metadata field registry
-defined in [TBD].  None of the fields defined below are intended to be mandatory
+If a JSON Object member value is an Ion Link, the Ion Link is implicitly assigned a link relation type that equals the member name.
+
+A non-normative example:
+
+[source,json]
+----
+{
+  "employer": { "meta": { "href": "https://api.example.com/corporations/acme" } }
+}
+----
+
+The above example shows an Ion Link member named `employer`.  The octet sequence `employer` is therefore assigned as the link's implicit link relation type.
```

Comment:
One thing that I think needs separating here is the application-level semantics from the structural semantics.

> But by having this in Ion, you ensure, to a stronger degree, that people won't break their domain model, because Ion is more resilient to data type changes over time than without this mandate. 

This should actually not be a problem dealt with in the structure of the document (i.e. the JSON). Of course you can restrict structures by saying only `items` may be an array, but you can also define areas in the document that can accommodate the change. Then, users can rely on client implementations to make that transparent. This is not possible to do with pure JSON, which is why I'm saying the JSON is the wrong place to hand-hold this kind of thing.

> When I thought of Ion, as I do (mostly) now, I don't want any specific domain semantics built in to the media type, but I did/do want to ensure that it would facilitate best practices of REST design

A question comes to mind: do you consider the Collection Resource to be a best practice of REST? This would be something good to chat about. I personally don't see it as important if link relations are used correctly. But I'd be interested in hearing your thoughts on why this is a best practice for REST and where this is used and defined in the world of REST or "REST."